File: src/Entity/Amis.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * Amis
 *
 * @ORM\Table(name="amis")
 * @ORM\Entity
 */
class Amis
{
    /**
     * @var int
     *
     * @ORM\Column(name="idAmis", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     * @Groups("post:read")
     */
    private $idamis;

    /**
     * @var string
     *
     * @ORM\Column(name="destinataire", type="string", length=20, nullable=false)
     * @Groups("post:read")
     */
    private $destinataire;

    /**
     * @var string
     *
     * @ORM\Column(name="expediteur", type="string", length=20, nullable=false)
     * @Groups("post:read")
     */
    private $expediteur;

    /**
     * @var int
     *
     * @ORM\Column(name="etatDemande", type="integer", nullable=false)
     * @Groups("post:read")
     */
    private $etatdemande;

    public function getIdamis(): ?int
    {
        return $this->idamis;
    }

    public function getDestinataire(): ?string
    {
        return $this->destinataire;
    }

    public function setDestinataire(string $destinataire): self
    {
        $this->destinataire = $destinataire;

        return $this;
    }

    public function getExpediteur(): ?string
    {
        return $this->expediteur;
    }

    public function setExpediteur(string $expediteur): self
    {
        $this->expediteur = $expediteur;

        return $this;
    }

    public function getEtatdemande(): ?int
    {
        return $this->etatdemande;
    }

    public function setEtatdemande(int $etatdemande): self
    {
        $this->etatdemande = $etatdemande;

        return $this;
    }


}

File: src/GestionAmis/entite/Amis.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package GestionAmis.entite;

public class Amis {
    private int idAmis;
    private String destinataire;
    private String expediteur;
    private int etatDemande;

    public Amis() {
    }

    public Amis(int idAmis, String destinataire, String expediteur, int etatDemande) {
        this.idAmis = idAmis;
        this.destinataire = destinataire;
        this.expediteur = expediteur;
        this.etatDemande = etatDemande;
    }

    public Amis(String destinataire, String expediteur) {
        this.destinataire = destinataire;
        this.expediteur = expediteur;
    }

    public int getIdAmis() {
        return idAmis;
    }

    public void setIdAmis(int idAmis) {
        this.idAmis = idAmis;
    }

    public String getDestinataire() {
        return destinataire;
    }

    public void setDestinataire(String destinataire) {
        this.destinataire = destinataire;
    }

    public String getExpediteur() {
        return expediteur;
    }

    public void setExpediteur(String expediteur) {
        this.expediteur = expediteur;
    }

    public int getEtatDemande() {
        return etatDemande;
    }

    public void setEtatDemande(int etatDemande) {
        this.etatDemande = etatDemande;
    }
    
    
    
}
